platform/ext/target/nordic_nrf/Native_Driver/spu.c

```diff
@@ -0,0 +1,164 @@
+
+#define FLASH_READ                                                             \
+	((SPU_FLASHREGION_PERM_READ_Enable << SPU_FLASHREGION_PERM_READ_Pos) & \
+	 SPU_FLASHREGION_PERM_READ_Msk)
+
+#define FLASH_SECURE                                                           \
+	((SPU_FLASHREGION_PERM_SECATTR_Secure                                  \
+	  << SPU_FLASHREGION_PERM_SECATTR_Pos) &                               \
+	 SPU_FLASHREGION_PERM_SECATTR_Msk)
+
+#define FLASH_NONSEC                                                           \
+	((SPU_FLASHREGION_PERM_SECATTR_Non_Secure                              \
+	  << SPU_FLASHREGION_PERM_SECATTR_Pos) &                               \
+	 SPU_FLASHREGION_PERM_SECATTR_Msk)
+
+#define SRAM_EXEC                                                              \
+	((SPU_RAMREGION_PERM_EXECUTE_Enable                                    \
+	  << SPU_RAMREGION_PERM_EXECUTE_Pos) &                                 \
+	 SPU_RAMREGION_PERM_EXECUTE_Msk)
+
+#define SRAM_WRITE                                                             \
+	((SPU_RAMREGION_PERM_WRITE_Enable << SPU_RAMREGION_PERM_WRITE_Pos) &   \
+	 SPU_RAMREGION_PERM_WRITE_Msk)
+
+#define SRAM_READ                                                              \
+	((SPU_RAMREGION_PERM_READ_Enable << SPU_RAMREGION_PERM_READ_Pos) &     \
+	 SPU_RAMREGION_PERM_READ_Msk)
+
+#define SRAM_SECURE                                                            \
+	((SPU_RAMREGION_PERM_SECATTR_Secure                                    \
+	  << SPU_RAMREGION_PERM_SECATTR_Pos) &                                 \
+	 SPU_RAMREGION_PERM_SECATTR_Msk)
+
+#define SRAM_NONSEC                                                            \
+	((SPU_RAMREGION_PERM_SECATTR_Non_Secure                                \
+	  << SPU_RAMREGION_PERM_SECATTR_Pos) &                                 \
+	 SPU_RAMREGION_PERM_SECATTR_Msk)
+
+#define FLASH_RWX (FLASH_READ | FLASH_WRITE | FLASH_EXEC)
+#define SRAM_RWX (SRAM_READ | SRAM_WRITE | SRAM_EXEC)
+
+
+#define FLASH_NSC_REGION(reg_number)                                           \
+	(((reg_number) << SPU_FLASHNSC_REGION_REGION_Pos) &                    \
+	SPU_FLASHNSC_REGION_REGION_Msk)
+
+#define FLASH_NSC_REGION_FROM_ADDR(addr)                                       \
+	FLASH_NSC_REGION(((u32_t)addr / FLASH_SECURE_ATTRIBUTION_REGION_SIZE))
+
+#define FLASH_NSC_SIZE(reg_size)                                               \
+	(((reg_size) << SPU_FLASHNSC_SIZE_SIZE_Pos) &                          \
+	SPU_FLASHNSC_SIZE_SIZE_Msk)
+
+#define FLASH_NSC_SIZE_FROM_ADDR(addr) FLASH_SECURE_ATTRIBUTION_REGION_SIZE    \
+	- (((u32_t)(addr)) % FLASH_SECURE_ATTRIBUTION_REGION_SIZE)
+
+#define FLASH_NSC_SIZE_REG(size) FLASH_NSC_SIZE((size) / FLASH_NSC_MIN_SIZE)
+
+
+void spu_regions_reset_all_secure(void)
+{
+    for (size_t i = 0; i < NUM_FLASH_SECURE_ATTRIBUTION_REGIONS ; i++) {
+            NRF_SPU->FLASHREGION[i].PERM = FLASH_RWX | FLASH_SECURE;
```

Comment:
Why not use `nrf_spu_flashregion_set` here and `nrf_spu_ramregion_set` right below?

Reply:
the reason is "for making the code more compressed". nrf_spu_flashregion_set would need to be called for each region.
the same is done (and for all targets) in `nvic_interrupt_target_state_cfg`

Reply:
I'm not sure I understand what you mean. `nrf_spu_flashregion_set` is an inline function, so the generated code will be the same (unless assertions are enabled).

Reply:
@anangl  `nrf_spu_flashregion_set` is inline, but contains two conditionals in the assignment operation (for secure attribute and lock booleans) which in certain cases would need to be evaluated in runtime, increasing the number of instructions executed. Most likely when passing constants, the compiler would optimize-out the '?' operators but this is not guaranteed by C code.

Reply:
Finally, fixed according to your proposal, as it allows me to get rid of the locally-defined macros for attributes

Reply:
@oyvindronningstad ^^